src/Compilers/CSharp/Portable/Binder/Binder_Attributes.cs

```diff
@@ -180,11 +180,20 @@ private BoundAttribute BindAttributeCore(AttributeSyntax node, NamedTypeSymbol a
             }
 
             var constructorArguments = analyzedArguments.ConstructorArguments;
-            ImmutableArray<BoundExpression> boundConstructorArguments = constructorArguments.Arguments.ToImmutableAndFree();
+
+            // If the arguments didn't have a natural type, they're not constants, and so would already have diagnostics from the rest of binding.
+            var ignoredDiagnostics = new DiagnosticBag();
+            ImmutableArray<BoundExpression> boundConstructorArguments = constructorArguments.Arguments.SelectAsArray(
+                (arg, thisAndDiagnostics) => thisAndDiagnostics.@this.BindToNaturalType(arg, thisAndDiagnostics.ignoredDiagnostics),
```

Comment:
>BindToNaturalType [](start = 70, length = 17)

AlekseyTs said:

> `BindToTypeForErrorRecovery`?
> We also have a `BuildArgumentsForErrorRecovery`, see if we can use that helper.

Reply:
Done. In order to use that helper, I had to split up the initialization into 2 versions: one where we could use the helper because we found a constructor, and one where we just bind to the natural type if it exists for the case where we didn't find a constructor.

---
In reply to: [575602126](https://github.com/dotnet/roslyn/pull/51206#discussion_r575602126) [](ancestors = 575602126)